repl/scala-2.12/src/main/scala/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -37,7 +37,14 @@ class SparkILoop(in0: Option[BufferedReader], out: JPrintWriter)
     @transient val spark = if (org.apache.spark.repl.Main.sparkSession != null) {
         org.apache.spark.repl.Main.sparkSession
       } else {
-        org.apache.spark.repl.Main.createSparkSession()
+        try {
+          org.apache.spark.repl.Main.createSparkSession()
+        } catch {
+          case e: Exception =>
+            println("Failed to initialize Spark session:")
+            e.printStackTrace()
+            sys.exit(1)
```

Comment:
Hm, I thought I had reverted this. Let me look again.

Reply:
yeah, looks like github is confused. It's definitely reverted in https://github.com/apache/spark/pull/21368/commits/a062e4c4c349e0035d50a6de9abc1e7eb04c1568

Reply:
how about just squashing the commits if it's not hard?

Reply:
Let me try that...